sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -242,6 +244,12 @@ def wait_for_idle(self, seconds_to_wait=None):
                 self.logger.error(error)
                 raise LivyClientTimeoutException(error)
 
+            if constants.YARN_RESOURCE_LIMIT_MSG in self._logs and \
+                not self._printed_resource_warning:
+                self.ipython_display.writeln(u"Warning: Session not starting due to resource limitation. {}"\
```

Comment:
This to me is not a very clear error message. What does "resource limitation" mean?